### PR TITLE
[CR] limits mass and volume that a internal furnace can consume

### LINKED ISSUE #25220: Internal furnace CBM allows you to consume objects larger than yourself
What it says on the tin. I'd recommend putting some sort of maximum mass or volume on items so players can't consume _objects larger than themselves_. Like skitterbots (heavier than many adults), or television sets.

### PR DESCRIPTION
**Pull request by PauloTP**
#### Summary

SUMMARY: Bugfixes "The internal furnace can only consume itens up to a maximum mass and volume"

#### Purpose of change

Fixes #25220 The player with the internal furnace CBM could consume massive itens, like tank drones

#### Describe the solution

Add a check to see if the volume of a item is larger than 3 liters and if the weight is higher than 6 kilograms (13.2277 pounds) if it is than the item doesn't appear into the possible consumables.

#### Describe alternatives you've considered

It could be better to check if all the itens materials are consumable, right now it only checks if one of the flags can be consumed, thus allowing itens like fire extinguishers and drones which are mostly made of metal to be consumed since they have the plastic tag.  

#### Additional context

I could use more information on what should be the maximum volume and weight allowed for the internal furnace, 3 litters and 30 kilograms allow as far as i know to consume most "normal" itens (from two by fours up to televisions) and it doesn't allow massive itens like skitterbots, logs and tank drones. 
Edit: changed to 6 Kg now.

### CODE REVIEW

**Comment by Xhuis** [comment]
I understand that, logically speaking, this is a good change, but *mechanically* speaking I'll always feel sad about it. Shoving entire wheels into my chest was something I always got a kick out of.

File: src/consumption.cpp [hunk 1]
```diff
@@ -52,6 +52,10 @@ const std::vector<std::string> carnivore_blacklist {{
 const std::array<std::string, 2> temparray {{"ALLERGEN_MEAT", "ALLERGEN_EGG"}};
 const std::vector<std::string> herbivore_blacklist( temparray.begin(), temparray.end() );
 
+//The maximum amount of mass and volume that a internal furnace can consume
+const units::volume max_volume (3000, units::volume_in_milliliter_tag{});
```
Comment (mlangsdorf) [inline src/consumption.cpp]: should be
```
+ const units::volume max_volume( 3000, units::volume_in_milliliter_tag{} );
```

File: src/consumption.cpp [hunk 2]
```diff
@@ -1069,6 +1079,7 @@ bool player::feed_furnace_with( item &it )
         return false;
     }
 
+
```
Comment (mlangsdorf) [inline src/consumption.cpp]: unnecessary white space.

File: src/consumption.cpp [hunk 3]
```diff
@@ -1059,6 +1063,10 @@ bool player::can_feed_furnace_with( const item &it ) const
     if( !has_active_bionic( bio_furnace ) ) {
         return false;
     }
+    
+    if((it.volume() > max_volume) || (it.weight() > max_mass)){
```
Comment (mlangsdorf) [inline src/consumption.cpp]: should be:
```
     if( ( it.volume() > max_volume ) || ( it.weight() > max_mass ) ) {
```
make sure you trim out any trailing spaces, too.

**Comment by alanbrady** [comment]
> This branch is 4 commits ahead, 480 commits behind CleverRaven:master

Looks like this might be an issue too and why gorgon is failing on files that are untouched here.  Needs to be rebased on master.  Just guessing, might be wrong.

**Comment by Kelenius** [comment]
I understand the volume restriction, but why should there be a weight restriction? If an item is somehow small but dense, that shouldn't be a problem for burning it.

**Comment by Formulos** [comment]
I did some more testing and i think limiting the consumption up to 3 L seems good, no need for weight restriction, tell me you you guys think.

File: src/consumption.cpp [hunk 4]
```diff
@@ -52,6 +52,9 @@ const std::vector<std::string> carnivore_blacklist {{
 const std::array<std::string, 2> temparray {{"ALLERGEN_MEAT", "ALLERGEN_EGG"}};
 const std::vector<std::string> herbivore_blacklist( temparray.begin(), temparray.end() );
 
+//The maximum amount of mass and volume that a internal furnace can consume
+const units::volume max_volume( 3000, units::volume_in_milliliter_tag{} );
```
Comment (BevapDin) [inline src/consumption.cpp]: You can write `3000_ml` when you need an instance of `units::volume` (it's called "user-defined literals").

Comment (Formulos) [reply src/consumption.cpp]: thank you, changed the name to furnace_max_volume

**Comment by Xhuis** [comment]
[Rest well, broken Internal Furnace. We'll remember you in the future.](https://www.youtube.com/watch?v=X2WH8mHJnhM)

**Pull request merged**